include/astarte_device_sdk/type.hpp

```diff
@@ -46,6 +48,50 @@ enum AstarteType : int8_t {
   kStringArray
 };
 
+/**
+ * @brief Convert a string to an AstarteType enum.
+ *
+ * @param typ The string representation of the astarte type.
+ * @return The corresponding AstarteType enum value, an error if the string is not a valid Astarte
+ * type.
+ */
+// TODO: use static or move function definition to .cpp file
```

Comment:
I think inline is fine.
For your info here is an interesting read on [inlining](https://isocpp.org/wiki/faq/inline-functions#inline-and-perf)